### PR TITLE
feat(quote): support quotes in preview mode

### PR DESCRIPTION
**Pull request by jh2oman**
Features:
* new `getQuoteInPreview` and `getQuoteOutPreview` methods

Relevant spec change: https://github.com/fiatconnect/specification/pull/70/files

### CODE REVIEW

**Review by jophish — COMMENTED** [review]
This looks fine to me, though wondering if you considered just adding new methods instead of overloading the current ones, e.g. `getQuoteInPreview`/`getQuoteOutPreview`. Presumably, clients will know when they want to get a preview or not, and as-is, clients won't be able to _know_ that the result from `createQuoteIn`/`createQuoteOut` contains a `quoteId` or not without manually checking.

File: test/fiat-connect-client.test.ts [hunk 1]
```diff
@@ -255,59 +309,76 @@ describe('FiatConnectClientImpl', () => {
...
-          method: 'POST',
-          headers: { 'Content-Type': 'application/json' },
-          body: JSON.stringify(mockQuoteRequestQuery),
-        }),
-      )
-      expect(response.isOk).toBeTruthy()
-      expect(response.unwrap()).toMatchObject(mockQuoteOutResponse)
+  describe('createQuoteIn', () => {
+    it('calls _createQuote with the expected parameters', async () => {
+      client._createQuote = jest.fn().mockResolvedValue('quote')
```
Comment (satish-ravi) [inline test/fiat-connect-client.test.ts:326]: optional nit: you can do `jest.spyOn(client, '_createQuote').mockResolvedValueOnce('quote')` and avoid creating the client every time in beforeEach

File: src/types.ts [hunk 2]
```diff
@@ -38,11 +39,17 @@ export interface FiatConnectApiClient {
...
   isLoggedIn(): boolean
   createQuoteIn(
-    params: QuoteRequestBody,
+    params: CreateQuoteParams,
   ): Promise<Result<QuoteResponse, ResponseError>>
   createQuoteOut(
-    params: QuoteRequestBody,
+    params: CreateQuoteParams,
   ): Promise<Result<QuoteResponse, ResponseError>>
+  createQuoteInPreview(
```
Comment (satish-ravi) [inline src/types.ts:47]: nit: should the preview be `get` instead of `create` since a quoteId is not "created"?

File: test/mocks.ts [hunk 3]
```diff
@@ -23,13 +23,21 @@ import {
...
 } from '@fiatconnect/fiatconnect-types'
-import { TransferRequestParams } from '../src/types'
+import { CreateQuoteParams, TransferRequestParams } from '../src/types'
 
 export const mockQuoteRequestQuery: QuoteRequestBody = {
   fiatType: FiatType.USD,
   cryptoType: CryptoType.cUSD,
   country: 'DE',
   address: '0x123',
+  preview: true,
```
Comment (satish-ravi) [inline test/mocks.ts:33]: nit:
```suggestion
  ...mockCreateQuoteParams,
  preview: true,
```

**Review by satish-ravi — APPROVED** [review]

**Pull request merged**